lib/fetch/index.js

```diff
@@ -274,8 +274,8 @@ function abortFetch (p, request, responseObject) {
 
   // 3. If request’s body is not null and is readable, then cancel request’s
   // body with error.
-  if (request.body != null) {
-    cancelBody(request.body, error)
+  if (request.body != null && isReadable(request.body?.stream)) {
+    request.body.stream.cancel(error)
```

Comment:
@szmarczak: We have another spec problem here... the stream might be locked at which point we are not allowed the call cancel... even though the stream is readable... ideas?

Reply:
@jasnell I think actually there might be a bug in node's implementation. It doesn't say anything about throwing invalid state if the stream is locked when cancelling? https://streams.spec.whatwg.org/#readable-stream-cancel

Reply:
```js
✖ Invalid state: ReadableStream is locked

  test: TAP
  at:
    line: 371
    column: 5
    file: node:internal/errors
    constructor: true
    function: NodeError
  stack: >
    EventEmitter.abortFetch (lib/fetch/index.js:292:26)

    EventTarget.requestObject.signal.addEventListener.once (lib/fetch/index.js:144:18)

    EventTarget.Request.signal.addEventListener.once (lib/fetch/request.js:335:16)
  type: TypeError
  code: ERR_INVALID_STATE
  tapCaught: unhandledRejection
```

Reply:
@szmarczak also... the spec doesn't say what to do if the promise returned from cancel rejects...

Reply:
> the stream might be locked at which point we are not allowed the call cancel

Locking just means it's "piped" to a reader. Cancellation still should be possible.

```js
const controller = new AbortController();
const response = await fetch('https://speed.hetzner.de/100MB.bin', {signal: controller.signal});
const reader = response.body.getReader();
console.log(response.body.locked);
controller.abort();
console.log(await reader.read()); // GoogleChrome - DOMException: The user aborted a request.
```

Reply:
> the spec doesn't say what to do if the promise returned from cancel rejects

I think it never rejects.